c/tests/integration/driver_test.c

```diff
@@ -0,0 +1,437 @@
+    options_drop(opts);
+
+    delete_database_if_exists(databaseManager, databaseName);
+    check_error_may_print(__FILE__, __LINE__);
+    database_manager_drop(databaseManager);
+    connection_close(connection);
+    return success;
+}
+
+bool test_concept_api_schema() {
+    const char databaseName[] = "test_concept_api";
+
+    Connection* connection = NULL;
+    DatabaseManager* databaseManager = NULL;
+    Session* session = NULL;
+    Transaction* transaction = NULL;
+    Options* opts = NULL;
+
+    bool success = false;
+
+    connection = connection_open_core(TYPEDB_CORE_ADDRESS);
+    if (FAILED()) goto cleanup;
+
+    databaseManager = database_manager_new(connection);
+    if (FAILED()) goto cleanup;
+
+    delete_database_if_exists(databaseManager, databaseName);
+    if (FAILED()) goto cleanup;
+
+    databases_create(databaseManager, databaseName);
+    if (FAILED()) goto cleanup;
+
+    opts = options_new();
+    if (FAILED()) goto cleanup;
+
+    // test schema api
+    {
+        session = session_new(databaseManager, databaseName, Schema, opts);
+        if (FAILED()) goto cleanup;
+
+        transaction = transaction_new(session, Write, opts);
+        if (FAILED()) goto cleanup;
+        {
+            Concept* definedNameType = concepts_put_attribute_type(transaction, "name", String);
+            if (FAILED()) goto cleanup;
+            else concept_drop(definedNameType);
+        }
+
+        {
+            ConceptIterator* it = NULL;
+            Concept* nameType = NULL;
+            Concept* rootAttributeType = NULL;
+            bool foundName = false;
+            if (
+                NULL != (nameType = concepts_get_attribute_type(transaction, "name")) &&
+                NULL != (rootAttributeType = concepts_get_attribute_type(transaction, "attribute")) &&
+                NULL != (it = attribute_type_get_subtypes(transaction, rootAttributeType, Transitive))) {
+                Concept* concept;
+                while (NULL != (concept = concept_iterator_next(it))) {
+                    char* label = thing_type_get_label(concept);
+                    foundName = foundName || (0 == strcmp(label, "name"));
+                    free(label);
+                    concept_drop(concept);
+                }
+            }
+            concept_iterator_drop(it);
+            concept_drop(rootAttributeType);
+            concept_drop(nameType);
+
+            transaction_commit(transaction);
+            transaction = NULL;
+            if (!foundName) {
+                fprintf(stderr, "Did not find type \'name\' in subtypes of attribute.\n");
+                goto cleanup;
+            }
+        }
+    }
+
+    success = true;
+
+cleanup:
+    transaction_drop(transaction);
+    session_drop(session);
+    options_drop(opts);
+
+    delete_database_if_exists(databaseManager, databaseName);
+    check_error_may_print(__FILE__, __LINE__);
+    database_manager_drop(databaseManager);
+    connection_close(connection);
```

Comment:
this should be `drop`

Reply:
connection, session, transaction and database now have `_close` instead of `_drop`